pkg/webui/oauth/views/update-password/index.js

```diff
@@ -0,0 +1,221 @@
+import queryString from 'query-string'
+
+import api from '../../api'
+import sharedMessages from '../../../lib/shared-messages'
+import { digit, uppercase, special } from '../../lib/regexp'
+
+import Button from '../../../components/button'
+import Form from '../../../components/form'
+import Input from '../../../components/input'
+import SubmitButton from '../../../components/submit-button'
+import IntlHelmet from '../../../lib/components/intl-helmet'
+import Message from '../../../lib/components/message'
+import style from '../create-account/create-account.styl'
+import Checkbox from '../../../components/checkbox'
+import Spinner from '../../../components/spinner'
+import PropTypes from '../../../lib/prop-types'
+
+const m = defineMessages({
+  newPassword: 'New Password',
+  oldPassword: 'Temporary or Old Password',
+  passwordChanged: 'Password changed successfully',
+  revokeAllAccess: 'Log out from all devices',
+  revokeWarning: 'This will revoke access from all logged in devices',
+})
+
+const validationSchema = Yup.object().shape({
+  password: Yup.string()
+    .min(8)
+    .matches(digit, sharedMessages.validatePasswordDigit)
+    .matches(uppercase, sharedMessages.validatePasswordUppercase)
+    .matches(special, sharedMessages.validatePasswordSpecial)
+    .required(sharedMessages.validateRequired),
+  confirm: Yup.string()
+    .oneOf([ Yup.ref('password'), null ], sharedMessages.validatePasswordMatch)
+    .min(8)
+    .required(sharedMessages.validateRequired),
+})
+
+const initialValues = {
+  password: '',
+  confirm: '',
+  password_changed: false,
+  revoke_all_access: false,
+}
+
+@connect(state => ({
+  fetching: state.user.fetching,
+  user: state.user.user,
+}), {
+  handleCancelUpdate: () => push('/'),
+  push,
+})
+@bind
+export default class UpdatePassword extends React.PureComponent {
+
+  static propTypes = {
+    fetching: PropTypes.bool.isRequired,
+    user: PropTypes.object,
+    handleCancelUpdate: PropTypes.func.isRequired,
+    push: PropTypes.func.isRequired,
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      error: '',
+      info: '',
+      revoke_all_access: false,
+    }
+  }
+
+  handleRevokeAllAccess (evt) {
+    this.setState({ revoke_all_access: evt.target.checked })
+  }
+
+  async handleSubmit (values, { setSubmitting }) {
+    const { user, push } = this.props
+    const userParams = queryString.parse(this.props.location.search)
+    const oldPassword = values.old_password ? values.old_password : userParams.current
+    const userId = Boolean(user) ? user.ids.user_id : userParams.user
+    try {
+      await api.users.updatePassword( userId, {
+        user_ids: { user_id: userId },
+        'new': values.password,
+        old: oldPassword,
+        revoke_all_access: values.revoke_all_access,
+      })
+      if (Boolean(user)) {
+        this.setState({
+          error: '',
+          info: m.passwordChanged,
+          changed: true,
+        })
+      } else {
```

Comment:
```suggestion
    resetForm(initialValues) 
      } else {
```
Best to clear the form after a successful update.